flow/middleware/interceptors.go

```diff
@@ -2,37 +2,51 @@ package middleware
 
 import (
 	"context"
+	"fmt"
 	"log/slog"
 	"time"
 
+	"github.com/hashicorp/golang-lru/v2/expirable"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/peer"
 	"google.golang.org/grpc/status"
 
 	"github.com/PeerDB-io/peer-flow/logger"
+	"github.com/PeerDB-io/peer-flow/peerdbenv"
 )
 
-func CreateAuthServerInterceptor(ctx context.Context, plaintext string, unauthenticatedMethods []string) grpc.UnaryServerInterceptor {
+func CreateAuthServerInterceptor(ctx context.Context, unauthenticatedMethods []string) (grpc.UnaryServerInterceptor, error) {
+	plaintext := peerdbenv.PeerDBPassword()
+
 	if plaintext == "" {
 		logger.LoggerFromCtx(ctx).Warn("Authentication is disabled")
 		return func(ctx context.Context, req any, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (any, error) {
 			return handler(ctx, req)
-		}
+		}, nil
 	}
+
 	unauthenticatedMethodsSet := make(map[string]struct{})
 	for _, method := range unauthenticatedMethods {
 		unauthenticatedMethodsSet[method] = struct{}{}
 	}
+
+	// accommodate live password changes and reduce time hash is in memory
+	hashCache := expirable.NewLRU[string, []byte](1, nil, 10*time.Minute)
+	_, err := getCachedHash(hashCache)
+	if err != nil {
+		return nil, fmt.Errorf("error generating hash: %w", err)
+	}
+
 	return func(ctx context.Context, req any, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (any, error) {
 		if _, ok := unauthenticatedMethodsSet[info.FullMethod]; ok {
 			return handler(ctx, req)
 		}
-		ctx, err := Authorize(ctx, plaintext)
+		ctx, err := authorize(ctx, hashCache)
 		if err != nil {
```

Comment:
Why do we need to look at live password changes?
Why not send the hash directly?